pacman.py

```diff
@@ -254,15 +253,59 @@ def is_aur(package):
         return False
 
 
-def pacman(flags, pkgs=[], eflgs=[], pacman_bin=__PACMAN_BIN):
+def contents_of(packages):
+    """Get list of files and directories that are owned by these packages"""
+    s = pacman("-Ql", packages)
+    if s["code"] != 0:
+        raise Exception("Failed to get contents: {0}".format(s["stderr"]))
+    ret = dict()
+    for x in s["stdout"].split('\n'):
+        space = x.find(' ')
+        package = x[:space+ 1]
+        file = x[space:]
+        if package not in ret:
+            ret[package] = []
+        ret[package].append(file)
+
+    if type(packages) == str:
+        return ret[packages] if packages in ret else None
+
+    return ret
+
+
+def owner_of(files, include_version=True):
+    """Get owner of these files and directories"""
+    s = pacman("-Qo", files)
+    if s["code"] != 0:
+        raise Exception("Failed to get owner: {0}".format(s["stderr"]))
+    owner_re = re.compile(r'(.+?) is owned by ([^ ]+) (.*)')
+    ret = dict()
+    for x in s["stdout"].split('\n'):
+        matches = owner_re.match(x)
+        if not matches:
+            continue
+        file = matches.group(1)
+        package = matches.group(2)
+        version = matches.group(3)
+        ret[file] = (package, version) if include_version else package
+
+    if type(files) == str:
+        return ret[files] if files in ret else None
+
+    return ret
+
+
+def pacman(flags, targets=[], eflgs=[], pacman_bin=__PACMAN_BIN):
```

Comment:
Renamed `pkgs` parameter to `targets`, since it also now holds file paths (`owner_of()`)